### ISSUE TITLE
Add deriveHasField to drop data constructor prefix

### ISSUE THREAD

**Issue by chiroptical** [chiroptical/derive-has-field#11]
```
data TypeConstructor = DataConstructor
  { dataConstructorX :: X
  }

deriveHasField ''TypeConstructor
```

would be equivalent to `deriveHasFieldWith (dropPrefix "dataConstructor") ''TypeConstructor`

**Comment by chiroptical** [comment]
I have no idea how common it would be to have asymmetric type and data constructors like this, but I took a poll at work and data constructor was considered the least surprising option.

**Comment by chiroptical** [comment]
Solved in #12
